### ISSUE TITLE
[FEATURE]: passing arguments which are not the result of previous steps into steps

### ISSUE THREAD

**Issue by kargarisaac** [zenml-io/zenml#869]
### Describe the feature you'd like

I really like to pass arguments which are not the output of previous steps to a new step. It's useful for ML models hyperparameter tuning. 

Right now I have to run the pipeline several times for several set of hyperparameters or use the trick in [this notebook](https://github.com/zenml-io/zenbytes/blob/main/2-1_Experiment_Tracking.ipynb) which again runs the pipeline several times. 

It would be great to just run previous steps once and have kind of a loop in the training step. I think that in this case I have to use mlflow manually to track data and cannot use `enable_mlflow` decorator. It will consider all of them as one run.

In summary I want to use a loop as follows and have the hyperparameters as input to the step:

```python
@step
def train_model(X_train, y_train, X_test, y_test, hypers):
        # X_train, y_train, X_test, y_test are from previous steps but hypers is not!
        for hyper in hypers:
                with mlflow.start_run():
                    ## model definition
                    model = tf.keras.Sequential([...])
                    ## training
                    model.compile(...)
                    mlflow.log_param(...)
                    model.fit(...)
                    mlflow.keras.log_model(...)
```

Thank you

**Comment by htahir1** [comment]
Basically you're looking for a tuner component, which does hyper-parameter tuning and tracks things in an experiment tracking stack component. We have this on our radar (https://zenml.hellonext.co/p/enable-hyper-parameter-tuning) and will get to it soon and potentially unlock this feature

**Comment by kargarisaac** [comment]
Thanks for the link. exactly :)

**Comment by htahir1** [comment]
I will close this issue as we are tracking it on Hellonext :-)
